src/crypto/scrypt.cpp

```diff
@@ -243,14 +244,16 @@ void scrypt_detect_sse2()
     if (cpuid_edx & 1<<26)
     {
         scrypt_1024_1_1_256_sp_detected = &scrypt_1024_1_1_256_sp_sse2;
-        printf("scrypt: using scrypt-sse2 as detected.\n");
+        fUsingSSE2 = true;
     }
     else
     {
         scrypt_1024_1_1_256_sp_detected = &scrypt_1024_1_1_256_sp_generic;
```

Comment:
This line looks redundant, unless there's a chance runtime CPU feature detection can change. I don't mind keeping it so that both branches of the conditional are parallel, but properties of the running system could change as a side effect of calling this function. Is there a measurable risk to consider here?

I know we don't want to change line 223, because that'll break the side effect property if this function takes the branch on line 227.

Or (my slight preference), do we want to skip this question until we get runtime intrinsic detection working reliably?

Reply:
In 3a1fd97f, I tried to not change any of the code that was working and do as little refactoring as possible. I agree with your assessment and I agree also that the best way to address it is by implementing the generic runtime capability detection rather than fixing this now - we're likely to remove this completely.

Reply:
We've just now documented our intent to rework this code in the near future, and that's good enough for me.